### PR TITLE
Added a function to visualise a spectra as a rainbow

### PR DESCRIPTION
**Pull request by stephenmwilkins**
DWISOTT

Example plots:
2f85091b-e594-4d4d-bb8e-8f51bfbd8f95
baf9f92e-606f-437f-a6c8-5c2ff5ad27cd

## Issue Type

- [x] Enhancement

## Checklist
- [x] I have read the [CONTRIBUTING.md]() -->
- [x] I have added docstrings to all methods
- [x] I have added sufficient comments to all lines
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no pep8 errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/synthesizer/sed.py [hunk 1]
```diff
@@ -1689,6 +1689,72 @@ def plot_observed_spectra(
     return fig, ax
 
 
+def plot_spectra_as_rainbow(
+    sed, figsize=(5, 0.5), lam_min=3000, lam_max=8000, include_xaxis=True
```
[the remark this answers is not in the document]
Comment (stephenmwilkins) [reply src/synthesizer/sed.py:1693]: TBH I'm not sure it's particularly useful to allow logging, and you have to add in a minimum as well.

[the remark this answers is not in the document]
Comment (WillJRoper) [reply src/synthesizer/sed.py:1693]: The minimum could definitely be automatically found but fair enough if you don't think it's worth while.

[the remark this answers is not in the document]
Comment (stephenmwilkins) [reply src/synthesizer/sed.py:1693]: I've done it anyway

File: src/synthesizer/sed.py [hunk 2]
```diff
@@ -1689,6 +1689,72 @@ def plot_observed_spectra(
...
+
+    Returns:
+        fig (matplotlib.pyplot.figure)
+            The matplotlib figure object for the plot.
+        ax (matplotlib.axes)
+            The matplotlib axes object containing the plotted data.
+    """
+
+    # define filter for spectra
+    s = (sed._lam < lam_max) & (sed._lam > lam_min)
```
[the remark this answers is not in the document]
Comment (stephenmwilkins) [reply src/synthesizer/sed.py:1718]: Done

File: src/synthesizer/sed.py [hunk 3]
```diff
@@ -1689,6 +1689,72 @@ def plot_observed_spectra(
     return fig, ax
 
 
+def plot_spectra_as_rainbow(
```
[the remark this answers is not in the document]
Comment (stephenmwilkins) [reply src/synthesizer/sed.py:1692]: Done

File: src/synthesizer/sed.py [hunk 4]
```diff
@@ -1689,6 +1689,72 @@ def plot_observed_spectra(
...
+    # get an array of colours
+    colours = np.array(
+        [wavelength_to_rgba(lam_, alpha=lnu_) for lam_, lnu_ in zip(lam, lnu)]
+    )
+
+    # expand dimensions to get an image array
+    im = np.expand_dims(colours, axis=0)
+
+    # show image
+    ax.imshow(im, aspect="auto", extent=[lam_min, lam_max, 0, 1])
```
[the remark this answers is not in the document]
Comment (stephenmwilkins) [reply src/synthesizer/sed.py]: Done

**Comment by stephenmwilkins** [comment]
Is there a method on `Sed` what wraps `plot_spectra`?

**Comment by WillJRoper** [comment]
> Is there a method on `Sed` what wraps `plot_spectra`?
Apparently not but there should be. There is on a `Galaxy` and `Components`.

**Comment by WillJRoper** [comment]
All a wrapper method needs to do is call the function passing self for the `sed`

**Comment by stephenmwilkins** [comment]
Done everything now.

**Comment by WillJRoper** [comment]
> Done everything now.

I think you misunderstood me 😅 the example plots you put on the slack, can you put those in the description of this PR instead of pointing to Slack.

**Comment by stephenmwilkins** [comment]
2f85091b-e594-4d4d-bb8e-8f51bfbd8f95
baf9f92e-606f-437f-a6c8-5c2ff5ad27cd

Kind pointless when I was posting them for others to see on Slack anyway.

**Comment by WillJRoper** [comment]
It's not at all pointless. The history for Synthesizer isn't contained within Slack (which will be cleared out every 90 days) it's here on GitHub. I wouldn't expect every person to ever use Synthesizer to join the Slack but they have the option to. This is the primary place.

**Review by WillJRoper — APPROVED** [review]

**Pull request merged**